raiden-ts/src/utils/error.ts

```diff
@@ -12,7 +12,58 @@ export type ErrorCodes = keyof typeof ErrorCodes;
 export const ErrorDetails = t.record(t.string, t.union([t.string, t.number, t.boolean, t.null]));
 export interface ErrorDetails extends t.TypeOf<typeof ErrorDetails> {}
 
-export const networkErrors: readonly string[] = ['invalid response'];
+export type ErrorMatch = string | number | { [k: string]: string | number };
+export type ErrorMatches = readonly ErrorMatch[];
+
+// overloads
+export function matchError(match: ErrorMatch | ErrorMatches): (error: any) => boolean;
+export function matchError(match: ErrorMatch | ErrorMatches, error: any): boolean;
+/**
+ * Matches an error, or creates a matcher for the error
+ * The matcher is curried on the left side, meaning that it'll return a function to check errors
+ * against the provided match or matches if error isn't provided as 2nd parameter.
+ * Matches can be strings (to be checked as substrings of error.message), numbers (to be checked
+ * for equality with error.httpStatus), or an arbitrary mapping of { key: values }, to check
+ * for strict property equality on the error object.
+ *
+ * @param match - Match or array of Matches to check
+ * @param error - Error to check
+ * @returns boolean if 'error' matches 'match' or some 'match', or a matcher function for 'error',
+ *   if 2nd param is undefined
+ */
+export function matchError(match: ErrorMatch | ErrorMatches, error?: any) {
+  const _errorMatcher = (match: ErrorMatch, error: any): boolean => {
+    return typeof match === 'string'
+      ? error?.message?.includes(match)
```

Comment:
Do we need the tertiary operators here for typing reasons? Otherwise some if-else would make this clearer.

Reply:
It was for typing, but since this was not assigning to a `const` variable and instead just `return`ed, it could be done as well with `if/else`. Implemented your suggestion, thanks.